x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucket.java

```diff
@@ -216,6 +216,24 @@ private TypeResolution resolveType(BiFunction<Expression, TypeResolutions.ParamO
         return isFoldable(to, sourceText(), FOURTH);
     }
 
+    public static TypeResolution isStringOrDate(Expression e, String operationName, TypeResolutions.ParamOrdinal paramOrd) {
+        return TypeResolutions.isType(
+            e,
+            exp -> { return DataTypes.isString(exp) || DataTypes.isDateTime(exp); },
+            operationName,
+            paramOrd,
+            "datetime",
+            "string"
+        );
+    }
+
+    private long convertToLong(Expression e) {
```

Comment:
IMO `convertToLong` suggests that the expression is going to be interpreted as long, not folded to long. It'd be good to convey that intention too. Maybe `foldToLong()`, `foldToMillis()` or `foldAsDatetime()` (it "just so happens" that datetime is encoded as a long) or along these lines.

Reply:
Renamed, thank you for the suggestion!